spidermon/contrib/stats/statscollectors.py

```diff
@@ -0,0 +1,39 @@
+import os
+import pickle
+from collections import deque
+
+from scrapy.statscollectors import StatsCollector
+from scrapy.utils.project import data_path
+
+
+class LocalStorageStatsHistoryCollector(StatsCollector):
+    def __init__(self, crawler):
+        super(LocalStorageStatsHistoryCollector, self).__init__(crawler)
+
+        statsdir = data_path("stats", createdir=True)
+        self.stats_location = os.path.join(
+            statsdir, "{}_stats_history".format(crawler.spider.name)
+        )
+
+    def open_spider(self, spider):
+        max_stored_stats = spider.crawler.settings.getint(
+            "SPIDERMON_MAX_STORED_STATS", default=100
+        )
+
+        if os.path.isfile(self.stats_location):
+            with open(self.stats_location, "rb") as stats_file:
+                _stats_history = pickle.load(stats_file)
+        else:
+            _stats_history = deque([], maxlen=max_stored_stats)
+
+        if _stats_history.maxlen != max_stored_stats:
+            _stats_history = deque([
+                stat for stat in _stats_history
```

Comment:
I believe you can simply use `deque(_stats_history, maxlen=max_stored_stats)`.

Reply:
I only tested in Python 3.6, though, maybe older versions do not support that.

Reply:
You are right. It worked in Python 2.7 as well.